packages/cli/src/metadataGeneration/metadataGenerator.ts

```diff
@@ -58,28 +59,9 @@ export class MetadataGenerator {
       }
 
       ts.forEachChild(sf, node => {
-        /**
-         * If we declare a namespace within a module, like we do in `tsoaTestModule.d.ts`,
-         * we need to explicitly get the children of the module declaration
-         * (`declare module 'tsoaTest'`) - which are the moduleBlock statements,
-         * because otherwise our type resolver cannot iterate over namespaces defined in that module.
-         */
-        if (ts.isModuleDeclaration(node)) {
-          /**
-           * For some reason unknown to me, TS resolves both `declare module` and `namespace` to
-           * the same kind (`ModuleDeclaration`). In order to figure out whether it's one or the other,
-           * we check the node flags. They tell us whether it is a namespace or not.
-           */
-          // eslint-disable-next-line no-bitwise
-          if ((node.flags & ts.NodeFlags.Namespace) === 0 && node.body && ts.isModuleBlock(node.body)) {
-            node.body.statements.forEach(statement => {
-              this.nodes.push(statement);
-            });
-            return;
-          }
+        if (node.kind === ts.SyntaxKind.ClassDeclaration && this.IsExportedNode(node as ts.ClassDeclaration) && getDecorators(node, identifier => identifier.text === 'Route').length) {
```

Comment:
```suggestion
        if (ts.isClassDeclaration(node) && getDecorators(node, identifier => identifier.text === 'Route').length) {
```

Seems like isExported is basically a noop, also ts now ships with type guards which is a lot easier to read.

Reply:
Haha Go and enjoy your holiday 😜 but thanks, i will fix that!